internal/syncronizedmap/syncronizedmap.go

```diff
@@ -0,0 +1,54 @@
+package syncronizedmap
+
+import "sync"
+
+//FIXME: add UT
+type SyncronizedKernelOsDtkMapping interface {
+	NewNodeInfo(kernelVersion, osImage string)
+	NewImageStreamInfo(osImage, dtkImage string)
+	GetImage(kernelVersion string) (string, bool)
+}
+
+type syncronizedKernelOsDtkMapping struct {
+	m          *sync.RWMutex
+	kernelToOs map[string]string
+	osToDtk    map[string]string
+}
+
+func NewSyncronizedKernelOsDtkMapping() SyncronizedKernelOsDtkMapping {
+
+	return &syncronizedKernelOsDtkMapping{
+		m:          &sync.RWMutex{},
+		kernelToOs: map[string]string{},
+		osToDtk:    map[string]string{},
+	}
+}
+
+func (skom *syncronizedKernelOsDtkMapping) NewNodeInfo(kernelVersion, osImage string) {
+
+	skom.m.Lock()
+	defer skom.m.Unlock()
+
+	skom.kernelToOs[kernelVersion] = osImage
+}
+
+func (skom *syncronizedKernelOsDtkMapping) NewImageStreamInfo(osImage, dtkImage string) {
+
+	skom.m.Lock()
+	defer skom.m.Unlock()
+
+	skom.osToDtk[osImage] = dtkImage
+}
+
+func (skom *syncronizedKernelOsDtkMapping) GetImage(kernelVersion string) (string, bool) {
+
+	skom.m.RLock()
+	defer skom.m.RUnlock()
+
+	osImage, ok := skom.kernelToOs[kernelVersion]
+	if !ok {
+		return "", false
```

Comment:
lets add an Error log here, since this function does not return error

Reply:
I though we don't want any error log in inner packages..

Reply:
in general we don't, since we stack them all in the returned errors. But in this case the error is not returned, so we do need some log in order to debug it

Reply:
Or we change this method to return an error?

Reply:
I prefer to keep the behavior as close as possible to a golang map access.
The caller of this method (`MakeBuildConfigTemplate`) will return a valid error in case this method return `false`.

Reply:
you have 2 options here to return false: kernel not present in the map or os not present in the map. Adding a log here will help to pinpoint the issue in case of an error

Reply:
As agreed with @yevgeny-shnaidman we will leave it as is.